rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -54,10 +54,11 @@ NodeGraph::get_topic_names_and_types() const
   auto ret = rcl_get_topic_names_and_types(node_base_->get_rcl_node_handle(),
       &topic_names_and_types);
   if (ret != RMW_RET_OK) {
-    // *INDENT-OFF*
-    throw std::runtime_error(
-      std::string("could not get topic names and types: ") + rmw_get_error_string_safe());
-    // *INDENT-ON*
+    auto error_msg = std::string("Failed to get topic names and types.");
+    if (rmw_destroy_topic_names_and_types(&topic_names_and_types) != RMW_RET_OK) {
+      error_msg += "Fatal: Leaking topic_names_and_types memory.";
```

Comment:
This would end up with a string like `Failed to get topic names and types.Fatal: Leaking topic_names_and_types memory.` which would be strange.

I think it's acceptable to throw from here, overruling the throw on the next line. Or you could throw here and reword it as `failed to cleanup after failing to get the topic names and types, leaking memory`.

Also, try to follow the existing style for error messages were possible. The message you're removing notably is not capitalized and lacks punctuation. This is also the case in other places in this file already.